chain/cmd/bandd/main.go

```diff
@@ -64,10 +71,122 @@ func GenFilePVIfNotExists(cdc *codec.Codec, keyFilePath, stateFilePath string) {
+
+	return printInfo{
+		Moniker:    moniker,
+		ChainID:    chainID,
+		NodeID:     nodeID,
+		GenTxsDir:  genTxsDir,
+		AppMessage: appMessage,
+	}
+}
+
+func displayInfo(cdc *codec.Codec, info printInfo) error {
+	out, err := codec.MarshalJSONIndent(cdc, info)
+	if err != nil {
+		return err
+	}
+
+	_, err = fmt.Fprintf(os.Stderr, "%s\n", string(sdk.MustSortJSON(out)))
+	return err
+}
+
+// InitCmd returns a command that initializes all files needed for Tendermint
+// and the respective application.
+func InitCmd(ctx *server.Context, cdc *codec.Codec, customAppState map[string]json.RawMessage,
+	defaultNodeHome string) *cobra.Command { // nolint: golint
+	cmd := &cobra.Command{
+		Use:   "init [moniker]",
+		Short: "Initialize private validator, p2p, genesis, and application configuration files",
+		Long:  `Initialize validators's and node's configuration files.`,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			config := ctx.Config
+			config.SetRoot(viper.GetString(cli.HomeFlag))
+
+			chainID := viper.GetString(client.FlagChainID)
+			if chainID == "" {
+				chainID = fmt.Sprintf("test-chain-%v", common.RandStr(6))
+			}
+
+			nodeID, _, err := genutil.InitializeNodeValidatorFiles(config)
+			if err != nil {
+				return err
+			}
+
+			config.Moniker = args[0]
+
+			genFile := config.GenesisFile()
+			if !viper.GetBool(flagOverwrite) && common.FileExists(genFile) {
+				return fmt.Errorf("genesis.json file already exists: %v", genFile)
+			}
+			appState, err := codec.MarshalJSONIndent(cdc, customAppState)
+			if err != nil {
+				return err
+			}
+
+			genDoc := &types.GenesisDoc{}
+			if _, err := os.Stat(genFile); err != nil {
+				if !os.IsNotExist(err) {
+					return err
+				}
+			} else {
+				genDoc, err = types.GenesisDocFromFile(genFile)
+				if err != nil {
+					return err
+				}
+			}
+
+			genDoc.ChainID = chainID
+			genDoc.Validators = nil
+			genDoc.AppState = appState
+			genDoc.ConsensusParams = &types.ConsensusParams{
+				Block: types.BlockParams{
+					MaxBytes:   200000, // 0.2MB
+					MaxGas:     2000000,
+					TimeIotaMs: 1000, // 1s
+				},
+				Evidence: types.EvidenceParams{
+					MaxAge: 100000, // 27.8 hrs at 1block/s
+				},
+				Validator: types.ValidatorParams{PubKeyTypes: []string{types.ABCIPubKeyTypeSecp256k1}},
+			}
+			if err = genutil.ExportGenesisFile(genDoc, genFile); err != nil {
+				return err
+			}
+
+			toPrint := newPrintInfo(config.Moniker, chainID, nodeID, "", appState)
+
+			cfg.WriteConfigFile(filepath.Join(config.RootDir, "config", "config.toml"), config)
+			return displayInfo(cdc, toPrint)
+		},
+	}
+
+	cmd.Flags().String(cli.HomeFlag, defaultNodeHome, "node's home directory")
+	cmd.Flags().BoolP(flagOverwrite, "o", false, "overwrite the genesis.json file")
+	cmd.Flags().String(client.FlagChainID, "", "genesis file chain-id, if left blank will be randomly created")
+
+	return cmd
+}
+
 func main() {
 	cobra.EnableCommandSorting = false
 
```

Comment:
Now that we write our own InitCmd. Can you also remove my hacky `PersistentPreRunE`? Basically revert my commit 7b41ef74081601ea0f3d9fb0930a2a13b4ee8287